File: tests/test_aes.py
import zlib

from Crypto.Cipher import AES


def unpack(file_path):
    key = b"1289@Patcher0000"
    iv = b"PatcherSDK000000"
    with open(file_path, "rb") as f:
        data = f.read()
    payload = data[16:]
    cipher = AES.new(key, AES.MODE_CBC, iv)
    dec = cipher.decrypt(payload)
    save(file_path, dec, "zlib")
    decompressor = zlib.decompressobj()
    xml = decompressor.decompress(dec)
    save(file_path, xml, "xml")


def save(original_path, final_data, file_ext):
    out_path = original_path + "." + file_ext
    with open(out_path, "wb") as f:
        f.write(final_data)


if __name__ == "__main__":
    unpack("ResList.bin")

File: tests/test_aes.rs
use nte_patcher::crypto::aes::unpack;
use std::io::Error;

#[test]
fn test_unpack_reslist() -> Result<(), Error> {
    let origin_path = "ResList.bin";
    let target_path = "ResList.xml";
    let key = b"1289@Patcher0000";
    let iv = b"PatcherSDK000000";
    unpack(origin_path, target_path, key, iv)?;
    Ok(())
}

#[test]
fn test_unpack_lastdiff() -> Result<(), Error> {
    let origin_path = "lastdiff.bin";
    let target_path = "lastdiff.xml";
    let key = b"1289@Patcher0000";
    let iv = b"PatcherSDK000000";
    unpack(origin_path, target_path, key, iv)?;
    Ok(())
}
